packages/core/content-manager/server/controllers/relations.js

```diff
@@ -132,54 +127,48 @@ module.exports = {
 
     await validateFindExisting(ctx.request.query);
 
-    const { component, ...query } = ctx.request.query;
-
-    const sourceModelUid = component || model;
-
-    const sourceModel = strapi.getModel(sourceModelUid);
-    if (!sourceModel) {
+    const modelSchema = strapi.getModel(model);
+    if (!modelSchema) {
       return ctx.badRequest("The model doesn't exist");
     }
 
-    const entityManager = getService('entity-manager');
-    const permissionChecker = getService('permission-checker').create({
-      userAbility,
-      model,
-    });
-
-    if (permissionChecker.cannot.read()) {
-      return ctx.forbidden();
-    }
-
-    const attribute = sourceModel.attributes[targetField];
+    const attribute = modelSchema.attributes[targetField];
     if (!attribute || attribute.type !== 'relation') {
       return ctx.badRequest("This relational field doesn't exist");
     }
 
-    // TODO: find a way to check field permission for component
-    if (!component && permissionChecker.cannot.read(null, targetField)) {
-      return ctx.forbidden();
-    }
+    const isComponent = modelSchema.modelType === 'component';
 
-    const entity = await entityManager.findOneWithCreatorRoles(id, model);
+    // RBAC checks when it's a content-type
+    // TODO: do RBAC check for components too
+    if (!isComponent) {
+      const entityManager = getService('entity-manager');
+      const permissionChecker = getService('permission-checker').create({
+        userAbility,
+        model,
+      });
 
-    if (!entity) {
-      return ctx.notFound();
-    }
+      if (permissionChecker.cannot.read(null, targetField)) {
+        return ctx.forbidden();
+      }
```

Comment:
Do you know if there is any documentation of the permissionChecker we could look?

Reply:
Not that I'm aware of. You can ask JS